spring-components/rdf4j-spring/src/main/java/org/eclipse/rdf4j/model/impl/MultiIRI.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * ******************************************************************************
+ *  * Copyright (c) 2021 Eclipse RDF4J contributors.
+ *  * All rights reserved. This program and the accompanying materials
+ *  * are made available under the terms of the Eclipse Distribution License v1.0
+ *  * which accompanies this distribution, and is available at
+ *  * http://www.eclipse.org/org/documents/edl-v10.php.
+ *  ******************************************************************************
+ */
+
+package org.eclipse.rdf4j.model.impl;
+
+import static org.eclipse.rdf4j.sparqlbuilder.rdf.Rdf.iri;
+
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.sparqlbuilder.rdf.Iri;
+
+/**
+ * @since 4.0.0
+ * @author Gabriel Pickl
+ */
+public class MultiIRI extends SimpleIRI implements Iri {
```

Comment:
I am actually not convinced this utility wrapper is a good idea. And it shouldn't be part of the core model, but part of the sparqlbuilder module (if at all). I'd suggest we remove this class from the PR as it's not really related to Spring.

Reply:
It was added because of the need to a) write vocabulary classes (like RDF.java) with minimal effort and to use these IRIs in the sparqlbuilder. It's possible that this is just an artifact created by our limited knowledge of the RDF4J API at the time.

I think removing it would be ok, but the issues we had are probably shared by others and we may want to think about how to ease the creation/use of vocabulary classes, especially with the sparqlbuilder. 

There are some related classes which you might want to review:
* `ExtendedPrefix.java` - If we remove `MultiIRI`, we should remove that, too
* `ExtendedVariable.java` - this one is very useful for retrieving values from query results - I would much like to keep it, or change `Variable` such that the alias can be publicly retrieved.
* `TypeMappingUtils.java` 
* `QueryResultUtils.java` - uses `ExtendedVariable`

Reply:
An example to illustrate the problem that `MultiIRI` solves:

`org.eclipse.rdf4j.sparqlbuilder.rdf.RdfSubject#isA` takes `RdfObject...` as its parameters. Now if I want to use my vocabulary there, I have to write eg `PAINTING_ID.isA(Rdf.iri(EX.Painting))`. With MultiIRI, this just works without having to remember to pass the IRI through `Rdf.iri`. 

An alternative (and maybe better than `MultiIRI` would be to overload such methods with one that takes `IRI` arguments and delegate them to the original method after conversion to an `Iri`, e.g.:
```
default TriplePattern isA(IRI... objects) {
		return isA(Arrays.stream(objects).map(Rdf::iri).collect(Collectors.toList()).toArray(new Iri[0]));
	}
```

Reply:
Oh you're quite right that there is a "disconnect" between the sparqlbuilder APIs and the model API, so I fully understand why you'd want something like this. In several other places in the SparqlBuilder we have in fact solved this by overloading, like you suggested. 

I'd just like to keep this question separate from this PR, that's all, and not introduce this new public class until we've thought it through. The curse of maintaining a public library using Semantic Versioning is that once we introduce a new class or interface to the API, we're stuck with it for a long time - so I like to be careful about what we introduce. 

If you do wish to keep it, we should at the very least annotate it using the `@Experimental` or `@InternalUseOnly` annotations - those are our "cop-out" annotations for warning people that we'll not honor semantic versioning requirements if we change our mind later.

Reply:
I removed MultiIRI and ExtendedPrefix from the PR. Probably won't come up again.